packages/context/src/invocation.ts

```diff
@@ -129,6 +131,12 @@ export type InvocationOptions = {
    * Skip invocation of interceptors
    */
   skipInterceptors?: boolean;
+  /**
+   * Information about the source object that makes the invocation. For REST,
+   * it's a `Route`. For injected proxies, it's a `Binding`.
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  source?: any;
```

Comment:
Can we avoid `any` please? It turns off compiler checks for any code that's reading `source`.

At minimum, I would prefer to use `unknown` to force consumers of `source` to deal with the fact that the actual source type is not known at compile time.

If we are sure that `Route` and `Binding` are the only possible types, then we can also replace `any` with `Route | Binding`, but I am not sure if that's feasible because here in `@loopback/context` we don't have access to `Route` from `@loopback/rest`.

On the second thought, this looks like a sign that our current design is no longer good enough and we need to improve it. Maybe the source should be always `Binding` and we should find a way how to access REST Route from its Binding?

I think the most important aspect is how are developers accessing `source` going to know what type to expect? For example, when writing a generic interceptor that can be applied to both controller methods and proxied methods, how is the author going to learn about different types they may receive in `source`?

I find the current design, where the type of  `source` is pretty much unknown or can be anything, as difficult to understand.